### PR TITLE
chore: refactored and added types for trading assessment

### PR DESCRIPTION
**Pull request by sanjam-deriv**
## Changes:

Please provide a summary of the change.

### Screenshots:

Please provide some screenshots of the change.

### CODE REVIEW

File: packages/account/src/Types/common.type.ts [hunk 1]
```diff
@@ -197,3 +197,25 @@ export type TServerError = {
...
+};
+
+export type TQuestion = {
+    question_text: string;
+    form_control: keyof TFormData;
+    answer_options: { text: string; value: string }[];
+    field_type?: string;
+};
+
+// export type TAssessmentQuestions = Array<TQuestion & { section?: string; questions: TQuestion[] }>;
```
Comment (matin-deriv) [inline packages/account/src/Types/common.type.ts]: You can remove the commented code.

File: packages/account/src/Constants/trading-assessment-questions.ts [hunk 2]
```diff
@@ -0,0 +1,252 @@
+import { localize } from '@deriv/translations';
+
+export const trading_assessment_questions = () => [
```
Comment (shaheer-deriv) [inline packages/account/src/Constants/trading-assessment-questions.ts]: `getTradingAssessmentQuestions = () => [ ....`

File: packages/account/src/Components/trading-assessment/trading-assessment-form.tsx [hunk 3]
```diff
@@ -17,16 +29,20 @@ const TradingAssessmentForm = ({
     should_move_to_next,
     setSubSectionIndex,
     is_independent_section,
-}) => {
+}: TTradingAssessmentForm) => {
     const [is_section_filled, setIsSectionFilled] = React.useState(false);
     const [should_inform_user, shouldInformUser] = React.useState(false);
+    const [form_data, setFormData] = React.useState<TFormData>({ ...form_value });
```
Comment (shaheer-deriv) [inline packages/account/src/Components/trading-assessment/trading-assessment-form.tsx]: do we need to specify `TFormData` explicitly? it will be inferred from `form_value`, right?
you may also ignore this.

File: packages/components/src/components/radio-group/radio-group.tsx [hunk 4]
```diff
@@ -8,10 +8,11 @@ type TItemWrapper = {
 };
 type TRadioGroup = {
     className?: string;
-    name: string;
+    name?: string;
```
Comment (shahzaib-deriv) [inline packages/components/src/components/radio-group/radio-group.tsx]: Is there a specific reason we're making `name` optional? I suppose `name` should always be required in a controlled form.

File: packages/account/src/Components/trading-assessment/trading-assessment-form.tsx [hunk 5]
```diff
@@ -102,15 +118,21 @@ const TradingAssessmentForm = ({
         }
     };
 
-    const handleValueSelection = (e, form_control, callBackFn) => {
+    const handleValueSelection = (
+        e: React.ChangeEvent<HTMLSelectElement>,
+        form_control: string,
```
Comment (shahzaib-deriv) [inline packages/account/src/Components/trading-assessment/trading-assessment-form.tsx]:
```suggestion
        form_control: keyof TFormData,
```

File: packages/account/src/Components/trading-assessment/trading-assessment-form.tsx [hunk 6]
```diff
@@ -17,16 +29,20 @@ const TradingAssessmentForm = ({
...
-}) => {
+}: TTradingAssessmentForm) => {
     const [is_section_filled, setIsSectionFilled] = React.useState(false);
     const [should_inform_user, shouldInformUser] = React.useState(false);
+    const [form_data, setFormData] = React.useState({ ...form_value });
+
+    const assessment_questions = trading_assessment_questions();
+
+    const stored_items = parseInt(localStorage.getItem('current_question_index') || '0');
+
```
Comment (shahzaib-deriv) [inline packages/account/src/Components/trading-assessment/trading-assessment-form.tsx]: stored_items is a bit ambiguous, can we rename to storage key `current_question_index`?

File: packages/account/src/Components/trading-assessment/trading-assessment-dropdown.tsx [hunk 7]
```diff
@@ -49,7 +65,7 @@ const TradingAssessmentDropdown = ({
                                         name={question?.question_text}
                                         placeholder={question?.question_text}
                                         list={question?.answer_options}
-                                        onChange={e => onChange(e, question.form_control, setFieldValue)}
+                                        onChange={e => onChange(e as any, question.form_control, setFieldValue)}
```
Comment (amina-deriv) [inline packages/account/src/Components/trading-assessment/trading-assessment-dropdown.tsx:80]: can we use eventType instead of any

File: packages/account/src/Components/trading-assessment/trading-assessment-new-user.tsx [hunk 8]
```diff
@@ -30,17 +48,19 @@ const TradingAssessmentNewUser = ({
                 return { ...accumulator };
             }, {});
         }
-        onSubmit(getCurrentStep() - 1, process_form_values, null, goToNextStep, should_override);
+        onSubmit(getCurrentStep() - 1, process_form_values, goToNextStep, null, should_override);
```
Comment (amina-deriv) [inline packages/account/src/Components/trading-assessment/trading-assessment-new-user.tsx]: why do we change this order of params

File: packages/account/src/Components/trading-assessment/trading-assessment-new-user.tsx [hunk 9]
```diff
@@ -30,17 +48,19 @@ const TradingAssessmentNewUser = ({
...
 
     return (
         <TradingAssessmentForm
-            assessment_questions={assessment_questions}
             form_value={value}
             onSubmit={handleSubmit}
             onCancel={handleCancel}
             setSubSectionIndex={setSubSectionIndex}
             disabled_items={disabled_items}
+            class_name={''}
```
Comment (amina-deriv) [inline packages/account/src/Components/trading-assessment/trading-assessment-new-user.tsx]: cant we make classname optional type in TradingAssessmentForm instead of passing empty classname here?

File: packages/account/src/Components/trading-assessment/__tests__/test-warning-modal.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+    it('shoukd render modal correctly on mobile', () => {
+        (isMobile as jest.Mock).mockReturnValue(true);
+        (isDesktop as jest.Mock).mockReturnValue(false);
+        const props = {
+            show_risk_modal: true,
+            title: 'Appropriateness Test Warning',
+            body_content: <div>body_content</div>,
+            footer_content: <div>fotter_content</div>,
+        };
```
Comment (suisin-deriv) [inline packages/account/src/Components/trading-assessment/__tests__/test-warning-modal.spec.tsx]: maybe we could move this const props out and share it between the 2 test cases

File: packages/account/src/Components/trading-assessment/trading-assessment-new-user.tsx [hunk 11]
```diff
@@ -30,17 +48,18 @@ const TradingAssessmentNewUser = ({
...
     return (
         <TradingAssessmentForm
-            assessment_questions={assessment_questions}
             form_value={value}
             onSubmit={handleSubmit}
             onCancel={handleCancel}
             setSubSectionIndex={setSubSectionIndex}
             disabled_items={disabled_items}
+            should_move_to_next={false}
+            is_independent_section={false}
```
Comment (suisin-deriv) [inline packages/account/src/Components/trading-assessment/trading-assessment-new-user.tsx]: Do we need to give false value here? if we don't pass in `should_move_to_next` and `is_independent_section` by default it will be false

File: packages/account/src/Components/trading-assessment/__tests__/trading-assessment-radio-button.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+    });
+
+    it('should render on the DOM', () => {
+        render(
+            <Formik initialValues={{}} onSubmit={jest.fn()}>
+                <Form>
+                    <TradingAssessmentRadioButton {...baseProps} />
+                </Form>
+            </Formik>
+        );
```
Comment (suisin-deriv) [inline packages/account/src/Components/trading-assessment/__tests__/trading-assessment-radio-button.spec.tsx:47]: Can we make this into a function

File: packages/account/src/Components/trading-assessment/risk-tolerance-warning-modal.tsx [hunk 13]
```diff
@@ -2,14 +2,23 @@ import React from 'react';
 import { Button, Icon, Modal, Text, DesktopWrapper, MobileDialog, MobileWrapper } from '@deriv/components';
 import { localize } from '@deriv/translations';
 
+type TRiskToleranceWarningModal = {
```
Comment (shaheer-deriv) [inline packages/account/src/Components/trading-assessment/risk-tolerance-warning-modal.tsx]: `TRiskToleranceWarningModalProps`

File: packages/account/src/Components/trading-assessment/risk-tolerance-warning-modal.tsx [hunk 14]
```diff
@@ -2,14 +2,23 @@ import React from 'react';
...
+
 const RiskToleranceWarningModal = ({
     show_risk_modal,
     handleAcceptRisk,
     title,
     button_text,
     body_content,
     has_sub_header = false,
-}) => {
+}: TRiskToleranceWarningModal) => {
```
Comment (shaheer-deriv) [inline packages/account/src/Components/trading-assessment/risk-tolerance-warning-modal.tsx]: => (<React.Fr...

File: packages/account/src/Components/trading-assessment/test-warning-modal.tsx [hunk 15]
```diff
@@ -2,7 +2,13 @@ import React from 'react';
 import { DesktopWrapper, MobileDialog, MobileWrapper, Modal } from '@deriv/components';
 import { localize } from '@deriv/translations';
 
-const TestWarningModal = ({ show_risk_modal, body_content, footer_content }) => (
+type TTestWarningModal = {
```
Comment (shaheer-deriv) [inline packages/account/src/Components/trading-assessment/test-warning-modal.tsx]: TTestWarningModalProps

File: packages/account/src/Components/trading-assessment/trading-assessment-dropdown.tsx [hunk 16]
```diff
@@ -3,7 +3,33 @@ import classNames from 'classnames';
 import { Field } from 'formik';
 import { DesktopWrapper, Dropdown, MobileWrapper, Text, SelectNative } from '@deriv/components';
 import { localize } from '@deriv/translations';
-import { MAX_QUESTION_TEXT_LENGTH } from '../../Constants/trading-assessment';
+import { TFormData, TQuestion } from 'Types';
+import { MAX_QUESTION_TEXT_LENGTH } from 'Constants/trading-assessment';
+
+type TTradingAssessmentDropdown = {
```
Comment (shaheer-deriv) [inline packages/account/src/Components/trading-assessment/trading-assessment-dropdown.tsx]: TTradingAssessmentDropdownProps

File: packages/account/src/Components/trading-assessment/trading-assessment-form.tsx [hunk 17]
```diff
@@ -2,15 +2,28 @@ import classNames from 'classnames';
...
-import TradingAssessmentDropdown from './trading-assessment-dropdown.jsx';
+import TradingAssessmentRadioButton from './trading-assessment-radio-buttons';
+import TradingAssessmentDropdown from './trading-assessment-dropdown';
+import { getTradingAssessmentQuestions } from 'Constants/trading-assessment-questions';
+import { TFormData, TQuestion } from 'Types';
+import ScrollToFieldWithError from 'Components/forms/scroll-to-field-with-error';
+import { MAX_QUESTION_TEXT_LENGTH } from 'Constants/trading-assessment';
+import { useStore } from '@deriv/stores';
+
+type TTradingAssessmentForm = {
```
Comment (shaheer-deriv) [inline packages/account/src/Components/trading-assessment/trading-assessment-form.tsx]: TTradingAssessmentFormProps

File: packages/account/src/Components/trading-assessment/__tests__/trading-assessment-form.spec.tsx [hunk 18]
```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom/extend-expect';
+import TradingAssessmentForm from '../trading-assessment-form';
+import { debug } from 'console';
```
Comment (amina-deriv) [inline packages/account/src/Components/trading-assessment/__tests__/trading-assessment-form.spec.tsx]: why do we need this?

File: packages/account/src/Components/trading-assessment/__tests__/trading-assessment-form.spec.tsx [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+        setSubSectionIndex: mockSetSubSectionIndex,
+        is_independent_section: false,
+    };
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should render without crashing', () => {
+        render(<TradingAssessmentForm {...baseProps} />);
```
Comment (amina-deriv) [inline packages/account/src/Components/trading-assessment/__tests__/trading-assessment-form.spec.tsx]: shouldnt we have a assertion here?

File: packages/account/src/Components/trading-assessment/__tests__/trading-assessment-new-user.spec.tsx [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+        },
+        setSubSectionIndex: jest.fn(),
+    };
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should render without crashing', () => {
+        render(<TradingAssessmentNewUser {...baseProps} />);
```
Comment (amina-deriv) [inline packages/account/src/Components/trading-assessment/__tests__/trading-assessment-new-user.spec.tsx]: assertion ?

File: packages/components/src/components/radio-group/radio-group.tsx (hunk 4 above)
Comment (sanjam-deriv) [reply packages/components/src/components/radio-group/radio-group.tsx]: Changed this

File: packages/account/src/Components/trading-assessment/trading-assessment-new-user.tsx (hunk 8 above)
Comment (sanjam-deriv) [reply packages/account/src/Components/trading-assessment/trading-assessment-new-user.tsx]: This is done to align with the type definition

File: packages/account/src/Components/trading-assessment/__tests__/trading-assessment-form.spec.tsx (hunk 19 above)
Comment (sanjam-deriv) [reply packages/account/src/Components/trading-assessment/__tests__/trading-assessment-form.spec.tsx]: removed the redudant test case

File: packages/account/src/Components/trading-assessment/__tests__/trading-assessment-new-user.spec.tsx (hunk 20 above)
Comment (sanjam-deriv) [reply packages/account/src/Components/trading-assessment/__tests__/trading-assessment-new-user.spec.tsx]: Removed the redunatant test case

File: packages/account/src/Components/trading-assessment/trading-assessment-dropdown.tsx (hunk 7 above)
Comment (sanjam-deriv) [reply packages/account/src/Components/trading-assessment/trading-assessment-dropdown.tsx:80]: Changed to eventType

File: packages/account/src/Components/trading-assessment/trading-assessment-form.tsx [hunk 21]
```diff
@@ -226,7 +228,11 @@ const TradingAssessmentForm = ({
                                                 text={question_text}
                                                 list={answer_options ?? []}
                                                 onChange={e => {
-                                                    handleValueSelection(e, form_control, setFieldValue);
+                                                    handleValueSelection(
+                                                        e as any,
```
Comment (amina-deriv) [inline packages/account/src/Components/trading-assessment/trading-assessment-form.tsx:232]: can we give actual type instead of any?

File: packages/account/src/Components/trading-assessment/__tests__/test-warning-modal.spec.tsx [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+        jest.clearAllMocks();
+    });
+
+    it('shoukd render modal correctly on desktop', () => {
+        (isDesktop as jest.Mock).mockReturnValue(true);
+        const props = {
+            show_risk_modal: true,
+            title: 'Appropriateness Test Warning',
+            body_content: <div>body_content</div>,
+            footer_content: <div>fotter_content</div>,
```
Comment (amina-deriv) [inline packages/account/src/Components/trading-assessment/__tests__/test-warning-modal.spec.tsx]: typo footer_content

File: packages/account/src/Components/trading-assessment/__tests__/test-warning-modal.spec.tsx (hunk 10 above)
Comment (amina-deriv) [reply packages/account/src/Components/trading-assessment/__tests__/test-warning-modal.spec.tsx]: is this done @sanjam-deriv

File: packages/account/src/Components/trading-assessment/trading-assessment-form.tsx [hunk 23]
```diff
@@ -19,15 +32,16 @@ const TradingAssessmentForm = ({
     should_move_to_next,
     setSubSectionIndex,
     is_independent_section,
-}) => {
+    is_mobile,
```
Comment (amina-deriv) [inline packages/account/src/Components/trading-assessment/trading-assessment-form.tsx:35]: cant we get this from usestore instead of prop passing

File: packages/components/src/components/radio-group/radio-group.tsx [hunk 24]
```diff
@@ -19,6 +19,7 @@ type TRadioGroup = {
     onToggle: (e: ChangeEvent<HTMLInputElement>) => void;
     required?: boolean;
     selected: string;
+    is_left?: boolean;
```
Comment (amina-deriv) [inline packages/components/src/components/radio-group/radio-group.tsx]: why is this added now? couldnt find its usage

File: packages/components/src/components/mobile-dialog/mobile-dialog.tsx [hunk 25]
```diff
@@ -15,7 +15,7 @@ type TMobileDialog = {
     has_full_height?: boolean;
     header_classname?: string;
     info_banner?: React.ReactNode;
-    onClose: React.MouseEventHandler;
+    onClose?: React.MouseEventHandler;
```
Comment (amina-deriv) [inline packages/components/src/components/mobile-dialog/mobile-dialog.tsx:18]: shouldnt onclose a required prop for dialogs

File: packages/components/src/components/contract-card/contract-card-items/vanilla-options-card-body.tsx [hunk 26]
```diff
@@ -48,9 +48,7 @@ const VanillaOptionsCardBody: React.FC<TVanillaOptionsCardBodyProps> = ({
                         {entry_spot_display_value && <Money amount={entry_spot_display_value} />}
                     </ContractCardItem>
 
-                    <ContractCardItem header={STRIKE}>
-                        {barrier && <Money amount={barrier} />}
-                    </ContractCardItem>
+                    <ContractCardItem header={STRIKE}>{barrier && <Money amount={barrier} />}</ContractCardItem>
```
Comment (amina-deriv) [inline packages/components/src/components/contract-card/contract-card-items/vanilla-options-card-body.tsx]: This and its test file changes are unrelated to this PR. lets revert this file changes

File: packages/account/src/Components/trading-assessment/trading-assessment-form.tsx (hunk 23 above)
Comment (sanjam-deriv) [reply packages/account/src/Components/trading-assessment/trading-assessment-form.tsx:35]: Since this not the main component which is being rendered , the store on the main component does not recognize the use store values from here and throws an here. So in order for store provider to recognize the useStore usage, passed it as a prop from the main component

File: packages/components/src/components/mobile-dialog/mobile-dialog.tsx (hunk 25 above)
Comment (sanjam-deriv) [reply packages/components/src/components/mobile-dialog/mobile-dialog.tsx:18]: `Risk-Tolerance-Warning-Modal` doesn't have onClose prop

File: packages/components/src/components/contract-card/contract-card-items/vanilla-options-card-body.tsx (hunk 26 above)
Comment (sanjam-deriv) [reply packages/components/src/components/contract-card/contract-card-items/vanilla-options-card-body.tsx]: This is just prettier changes
Do you want me to revert this? @amina-deriv

Comment (amina-deriv) [reply packages/components/src/components/contract-card/contract-card-items/vanilla-options-card-body.tsx]: lets revert them
